src/forcing/ForcingsEngine.cpp

```diff
@@ -0,0 +1,239 @@
+#include "ForcingsEngine.hpp"
+
+namespace data_access {
+
+std::unordered_map<std::string, std::unique_ptr<ForcingsEngine>> ForcingsEngine::instances_{};
+
+ForcingsEngine::ForcingsEngine(const std::string& init, size_type time_start, size_type time_end)
+  : time_start_(std::chrono::seconds{time_start})
+  , time_end_(std::chrono::seconds{time_end})
+{
+    check_runtime_requirements();
+
+    bmi_ = std::make_unique<models::bmi::Bmi_Py_Adapter>(
+        "ForcingsEngine",
+        init,
+        "NextGen_Forcings_Engine.BMIForcingsEngine",
+        true,
+        true,
```

Comment:
We should maybe change the constructor for `Bmi_Py_Adapter` to take enums instead of `bool`, but for now, could you add a small comment on each of these to denote what formal parameter the `true` literals correspond to?